applications/acoustic_conservation_equations/normal_incidenting_wave/application.h

```diff
@@ -0,0 +1,239 @@
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef APPLICATIONS_ACOUSTIC_CONSERVATION_EQUATIONS_TEST_CASES_NORMAL_INCIDENTING_WAVE_H_
+#define APPLICATIONS_ACOUSTIC_CONSERVATION_EQUATIONS_TEST_CASES_NORMAL_INCIDENTING_WAVE_H_
+
+#include <deal.II/base/function.h>
+#include <deal.II/distributed/tria.h>
+#include <deal.II/grid/grid_generator.h>
+
+#include <exadg/grid/grid_utilities.h>
+
+namespace ExaDG
+{
+namespace Acoustics
+{
+template<int dim>
+class InitialConditionPressure : public dealii::Function<dim>
+{
+public:
+  InitialConditionPressure() : dealii::Function<dim>(1, 0.0)
+  {
+  }
+
+  double
+  value(dealii::Point<dim> const & p, unsigned int const) const final
+  {
+    double exponent = 0.0;
+    for(unsigned int d = 0; d < dim; ++d)
+      exponent += p[d] * p[d];
+
+    return std::exp(-100.0 * exponent);
+  }
+};
+
+template<int dim, typename Number>
+class Application : public ApplicationBase<dim, Number>
+{
+public:
+  Application(std::string input_file, MPI_Comm const & comm)
+    : ApplicationBase<dim, Number>(input_file, comm)
+  {
+  }
+
+  void
+  add_parameters(dealii::ParameterHandler & prm) final
+  {
+    ApplicationBase<dim, Number>::add_parameters(prm);
+
+    prm.enter_subsection("Application");
+    {
+      prm.add_parameter("SpeedOfSound",
+                        speed_of_sound,
+                        "Speed of sound.",
+                        dealii::Patterns::Double(1.0e-12));
+
+      prm.add_parameter("Density", density, "Density.", dealii::Patterns::Double(1.0e-12));
+
+      prm.add_parameter("Admittance",
+                        admittance,
+                        "Admittance.",
+                        dealii::Patterns::Double(0.0, 1.0));
+
+      prm.add_parameter("EndTime", end_time, "End time.", dealii::Patterns::Double(1.0e-12));
+
+      prm.add_parameter("Radius", radius, "Radius of domain.", dealii::Patterns::Double(1.0e-12));
+    }
+    prm.leave_subsection();
+
+    // For a reasonable test we have to create a differen postprocessor, see below.
+    prm.enter_subsection("General");
+    {
+      prm.add_parameter("IsTest", is_test, "Do we run a test?", dealii::Patterns::Bool());
+    }
+    prm.leave_subsection();
```

Comment:
I am not sure whether we should use the existing variable `is_test` for this. I think I would go for an application-specific variable like `analytical_solution_is_available` or `use_first_order_admittance_bc`.

Reply:
[ce3f9a4](https://github.com/exadg/exadg/pull/629/commits/ce3f9a4cff7cfd9db600da22d60e3046c461f9e7)